lib/galaxy/webapps/galaxy/api/remote_files.py

```diff
@@ -132,20 +132,18 @@ def __create_jstree( self, directory, disable='folders' ):
         jstree_paths = []
         if os.path.exists( directory ):
             for ( dirpath, dirnames, filenames ) in os.walk( directory ):
-
                 for dirname in dirnames:
                     dir_path = os.path.relpath( os.path.join( dirpath, dirname ), directory )
-                    dir_path_hash = hashlib.sha1( dir_path.encode('utf-8') ).hexdigest()
+                    dir_path_hash = hashlib.sha1(dir_path).hexdigest()
                     disabled = True if disable == 'folders' else False
                     jstree_paths.append( jstree.Path( dir_path, dir_path_hash, { 'type': 'folder', 'state': { 'disabled': disabled }, 'li_attr': { 'full_path': dir_path } } ) )
 
                 for filename in filenames:
                     file_path = os.path.relpath( os.path.join( dirpath, filename ), directory )
-                    file_path_hash = hashlib.sha1( file_path.encode('utf-8') ).hexdigest()
+                    file_path_hash = hashlib.sha1(file_path).hexdigest()
```

Comment:
@dannon After a bit more investigations I've found out that the reason for the difference is that when `target=userdir`, `full_import_dir` is created by joining `user_login` (which is unicode) with `user_base_dir` (str). This obviously does not happen when `target=importdir`.

Also I can confirm that the version in my gist is the only one working on both str and unicode and in both Python 2 and 3.